enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,136 @@
+---
+title: SHA-Bundle-Digest
+authors:
+  - '@theishshah'
+reviewers:
+  - '@camilamacedo86'
+  - '@joelanford'
+  - '@jmrodri'
+  - '@ryantking'
+  - '@tlwu2013'
+approvers:
+  - '@tlwu2013'
+  - '@jmrodri'
+  - '@joelanford'
+
+creation-date: 2022-01-14
+
+last-updated: 2022-01-14
+
+status: implementable
+---
+
+# Support SHA-based Image Tags
+
+## Summary
+
+Currently, operators generated by `operator-sdk` will store and retrieve container images based on the image name and
+the image tag in the bundle. This EP adds support to `operator-sdk` such that images can be stored and retrieved in the
+bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that
+are hashed from the image layer therefore uniquely identifying one specific image.
```

Comment:
```suggestion
Operators built with `operator-sdk` use Docker images that are referenced via an image URL and tag name. This EP adds
the ability to create operators that use image digests instead of tags for operator images as well as related images. An
image digest is a `SHA-256` sum that is created by hashing the contents of a Docker image manifest. A Docker image
manifest is a JSON file that defines the image, most notably its layers. See [Appendix A](##appendix-a) for an example
manifest for `golang:1.17.6`. An alternative method to specify an image is to use its digest directly instead of a tag.
For example, an image for go 1.17.6 would be pulled with the Docker command `docker pull golang:1.17.6`. The exact same
image can be pulled by its digest as well:
`docker pull golang@sha256:197db40b76cce7e70d59e59cc567896395730c6491e7b3da5d1f4c22efa72404`. An important distinction
is that since digests are a hash of the image manifests, any change to the image, will cause it to have a distinct
digest. This means that if a new image is tagged as `golang:1.17.6`, the first `docker pull` command will pull that new
image while the second `docker pull` command will return the same image as before since it uses the digest to reference
it.
```